### PR TITLE
 Bring back DomainNameRoleInstanceTelemetryInitializer to fix bug 671

### PR DESCRIPTION
**Pull request by cijothomas**
Fix Issue #671  .
Brought back DomainNameRoleInstanceTelemetryInitializer 

- [ ] I ran Unit Tests locally.

For significant contributions please make sure you have completed the following items:

- [ ] Changes in public surface reviewed
- [ ] Design discussion issue #
- [ ] CHANGELOG.md updated with one line description of the fix, and a link to the original issue.
- [ ] The PR will trigger build, unit tests, and functional tests automatically. If your PR was submitted from fork - mention one of committers to initiate the build for you.
	  If you want to to re-run the build/tests, the easiest way is to simply Close and Re-Open this same PR. (Just click 'close pull request' followed by 'open pull request' buttons at the bottom of the PR)

- Please follow [these] (https://github.com/Microsoft/ApplicationInsights-aspnetcore/blob/develop/Readme.md) instructions to build and test locally.

### CODE REVIEW

File: src/Microsoft.ApplicationInsights.AspNetCore/TelemetryInitializers/DomainNameRoleInstanceTelemetryInitializer.cs [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+        /// <param name="telemetry">Telemetry item.</param>
+        public void Initialize(ITelemetry telemetry)
+        {
+            if (string.IsNullOrEmpty(telemetry.Context.Cloud.RoleInstance))
+            {
+                var name = LazyInitializer.EnsureInitialized(ref this.roleInstanceName, this.GetMachineName);
+                telemetry.Context.Cloud.RoleInstance = name;
+            }
+
+            InternalContext internalContext = telemetry.Context.GetInternalContext();
```
Comment (SergeyKanzhelev) [inline src/Microsoft.ApplicationInsights.AspNetCore/TelemetryInitializers/DomainNameRoleInstanceTelemetryInitializer.cs]: do not set nodeName if you set roleInstance. No need.

CC: @iusafaro

Comment (SergeyKanzhelev) [reply src/Microsoft.ApplicationInsights.AspNetCore/TelemetryInitializers/DomainNameRoleInstanceTelemetryInitializer.cs]: No, `nodename` will be set by `RoleInstance` on back end if not set explicitly. Only reason to set it on the environments that has `RoleInstance` name unstable and we want something more constant for billing.

Also nodename is only for legacy customers with existing contracts. No new customers benefit from having nodename. Thus it's ideal to not send it when possible to minimize the footprint

**Comment by cijothomas** [comment]
@SergeyKanzhelev got it.. i removed nodename.

I still have question about this line
https://github.com/Microsoft/ApplicationInsights-dotnet/blob/develop/src/Microsoft.ApplicationInsights/TelemetryClient.cs#L509

This will anyway populate nodename in windows, right?

**Comment by SergeyKanzhelev** [comment]
@cijothomas only if `!string.IsNullOrEmpty(telemetry.Context.Cloud.RoleInstance))` - one line above

**Comment by cijothomas** [comment]
@SergeyKanzhelev With this initializer in place, RoleInstance will be populated, but nodename will not be. And base sdk will see that Nodename is empty, and roleinstance is NOT empty, and so it will set the nodename.
`!string.IsNullOrEmpty(telemetry.Context.Cloud.RoleInstance))` this condition will be met since roleinstance is not empty.

btw - the comment is addressed please approve the PR if it look okay . The above discussion is mainly for my education :)

**Pull request merged**